cranelift/codegen/src/nan_canonicalization.rs

```diff
@@ -65,16 +65,23 @@ fn add_nan_canon_seq(pos: &mut FuncCursor, inst: Inst) {
     let new_res = pos.func.dfg.replace_result(val, val_type);
     let _next_inst = pos.next_inst().expect("block missing terminator!");
 
-    // Insert a comparison instruction, to check if `inst_res` is NaN. Select
-    // the canonical NaN value if `val` is NaN, assign the result to `inst`.
-    let is_nan = pos.ins().fcmp(FloatCC::NotEqual, new_res, new_res);
-
-    let scalar_select = |pos: &mut FuncCursor, canon_nan: Value| {
-        pos.ins()
-            .with_result(val)
-            .select(is_nan, canon_nan, new_res);
+    let scalar_select = |pos: &mut FuncCursor, canon_nan: Value, ty: types::Type| {
+        let canon_nan = pos.ins().scalar_to_vector(ty, canon_nan);
+        let new_res = pos.ins().scalar_to_vector(ty, new_res);
```

Comment:
I'm not sure this works with the RISC-V backend, our vector support is gated on a feature flag which may not always be available.

One solution would be to keep the previous code around for RISC-V, but it feels wrong to select on targets at this level.

Reply:
Ah I see that now. What about choosing based on the targets and features (that is specifically `riscv64` when `has_v=false`)? Maybe in the future there will be other target/feature combinations that don't support vectors?

Reply:
I think that is what we need right now. I'm not sure if eventually we should add something to `TargetIsa`, but It might be worth reviewing later on if this situation ends up happening more often.

Reply:
Added the fall back to the old version for RiscV [here](https://github.com/bytecodealliance/wasmtime/pull/8313/files#diff-c7a66b91ac03843c5aafe984938022ccba235c80c3fad786772964dc7b9da152R286).